cli/src/main/java/org/eclipse/hono/cli/application/Receiver.java

```diff
@@ -0,0 +1,131 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.cli.application;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+
+import org.apache.qpid.proton.message.Message;
+import org.eclipse.hono.cli.AbstractCommand;
+import org.eclipse.hono.client.ApplicationClientFactory;
+import org.eclipse.hono.client.HonoConnection;
+import org.eclipse.hono.util.MessageHelper;
+
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+
+
+/**
+ * The methods of the command line client for receiving messages from via Hono's north bound Telemetry and/or Event API
+ * <p>
+ * Messages are output to stdout.
+ * <p>
+ * Note that this example intentionally does not support Command &amp; Control and rather is the most simple version of a
+ * receiver for downstream data. Please refer to the documentation of Command &amp; Control for the example that supports
+ * it (found in the User Guide section).
+ */
+public class Receiver extends AbstractCommand {
+
+    private final ApplicationClientFactory clientFactory;
+    private final String tenantId;
+    private final String messageType;
+    private final int connectionRetryInterval;
+
+    /**
+     * Constructor to create the config environment for the execution of the command.
+     *
+     * @param clientFactory The factory with client's methods.
+     * @param vertx The instance of vert.x connection.
+     * @param tenantId The tenant which the user want to connect.
+     * @param messageType The message type which the user want to receive.
+     * @param connectionRetryInterval The time to wait before retry to connect.
+     */
+    public Receiver(final ApplicationClientFactory clientFactory, final Vertx vertx, final String tenantId, final String messageType, final int connectionRetryInterval) {
+        this.clientFactory = clientFactory;
+        this.vertx = vertx;
+        this.tenantId = tenantId;
+        this.messageType = messageType;
+        this.connectionRetryInterval = connectionRetryInterval;
+    }
+
+    /**
+     * Entrypoint to start the command.
+     *
+     * @param latch The handle to signal the ended execution and return to the shell.
+     * @param con The Hono connection instance, already connected.
+     */
+    public void start(final CountDownLatch latch, final HonoConnection con) {
+        this.latch = latch;
+        clientFactory.isConnected().setHandler(connectionStatus -> {
+            if (connectionStatus.succeeded()) {
+                clientFactory.addReconnectListener(this::createConsumer);
+                createConsumer(con);
+            } else {
+                close(connectionStatus.cause());
+            }
+        });
+    }
+
+    private CompositeFuture createConsumer(final HonoConnection connection) {
```

Comment:
what is the connection parameter good for?

Reply:
since the factory has a paramenter

` void addReconnectListener(ReconnectListener<T> listener);`